### PR TITLE
vim, MacVim: update to vim version 9.1.0000

### PR DESCRIPTION
**Pull request by thomasnyman**
#### Description

- update Vim to v9.1.0000
- update MacVim to release-179
- remove fallback version for macOS 10.10 and earlier (https://trac.macports.org/ticket/67209)

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1 23C71 arm64
Xcode 15.2 15C500b

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
   - Unable to run with trace-mode due to [#66356](https://trac.macports.org/ticket/66358)
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?
    - Tested Vim with `+cscope`, `+huge`, `+python311`, `+tcl`
    - Tested MacVim with `+cscope`, `+huge`, `+python311`, `+tcl`
    

###### Notes

- Vim patches `patch-python.diff` and `patch-tcl.diff` apply cleanly without changes. Patch `patch-python3.diff` not needed for v9.1.0000 ~but must be retained for fallback port version 9.0.0472 on MacOS 10.10 and earlier~. 
- MacVim patches `patch-python.diff`, `patch-remove-updater.diff`, `patch-remove-Homebrew-python.diff`, and `patch-tcl.diff` apply cleanly without changes.
- ~`vim_version` needs to now be set conditionally depending on MacOS version as the fallback version is one minor version behind the latest release.~

### CODE REVIEW

**Review by raimue — COMMENTED** [review]
Thank you for the update and the testing of the variants. Despite the no longer needed fallback this looks good so far from my side.

**Comment by thomasnyman** [comment]
@pmetzger I've pushed 211a1649650f3bddbbfdc2f0981c12ce407bb0c6 with the changes @raimue requested.

**Comment by thomasnyman** [comment]
> @thomasnyman The removal of the fallback also applies to patch-python3.diff and the conditional with which it is applied in the +pythonXY variant.

Thanks. That of course slipped my mind after a couple of days. The conditional and the obsolete `patch-python3.diff` are now removed in  23728e1489bd93cbded0af5cb7fa713edc098f3e.

File: editors/vim/Portfile [hunk 1]
```diff
@@ -6,25 +6,13 @@ PortGroup           github 1.0
 # The vim port should always follow the patch level used in the MacVim port. This
 # is because they both share the same set of configuration files and ensures
 # that the user's configuration files will always work with both.
-set vim_version     9.0
+set vim_version    9.1
+set vim_patchlevel 0000
+set port_revision  0
```
Comment (raimue) [inline editors/vim/Portfile:11]:
```suggestion
```

File: editors/vim/Portfile [hunk 2]
```diff
@@ -6,25 +6,13 @@ PortGroup           github 1.0
...
-    checksums       rmd160  03c9bc4d54789cbec843fbfbfa342de2f6af4384 \
-                    sha256  8b86cfb2b609e7a7e386b88a89b01522354990d4eef27e7001cc607d49d62ebf \
-                    size    17268335
-}
+checksums           rmd160  aed3a40e2fb0a96fc22c554b2fd102fb3369d560 \
+                    sha256  f0cdc73963122d052a666d88f787ec5b295f75ebdc3dfa5aa5a48f8afa92d98f \
+                    size    17604158
 
 github.setup        vim vim ${vim_version}.${vim_patchlevel} v
 revision            ${port_revision}
```
Comment (raimue) [inline editors/vim/Portfile:18]:
```suggestion
revision            0
```

**Review by raimue — COMMENTED** [review]
Sorry, only spotted this now. But we should also get rid of the `port_revision` variable to keep the usual syntax.

File: editors/vim/Portfile (hunk 2 above)
Comment (thomasnyman) [reply editors/vim/Portfile]: I've removed the `port_revision` variable in 3809d06b514ed9c36b87457d4e0b62940381f47c.

I also moved the `checksums` section later to be consistent with the layout of the MacVim port.

**Comment by pmetzger** [comment]
@raimue Looks like this can now be merged?

**Review by raimue — APPROVED** [review]
@pmetzger Yes, looks good to me now.
@thomasnyman Thanks again for the port update!

**Pull request merged**